archaius2-core/src/main/java/com/netflix/archaius/readers/PropertiesConfigReader.java

```diff
@@ -64,16 +64,16 @@ private void internalLoad(Properties props, Set<String> seenUrls, ClassLoader lo
             try {
                 // Load properties into the single Properties object overriding any property
                 // that may already exist
-                Map<String, String> p = new URLConfigReader(url).call().getToAdd();
+                Map<String, Object> p = new URLConfigReader(url).call().getToAdd();
                 LOG.debug("Loaded : {}", url.toExternalForm());
                 props.putAll(p);
     
                 // Recursively load any files referenced by an @next property in the file
                 // Only one @next property is expected and the value may be a list of files
-                String next = p.get(INCLUDE_KEY);
-                if (next != null) {
+                Object next = p.get(INCLUDE_KEY);
+                if (next != null && next instanceof String) {
```

Comment:
Next should only ever be a string.  It's OK to keep next as a String.

String next = (String)p.get(INCLUDE_KEY);